app/models/web_authenticator.rb

```diff
@@ -1,6 +1,14 @@
-class WebAuthenticator < Struct.new(:controller)
+class WebAuthenticator < Struct.new(:controller, :params)
   def current_user
-    @current_user ||= User.where(id: controller.session[:user_id]).first || Guest.new
+    if @current_user.nil?
+      @current_user = if params[:token]
+                        GuestWithToken.new(params[:token])
```

Comment:
so if the token is invalid, this line will cause an AR::RecordNotFound to be raised. I don't think that will be rescued anywhere. Is that ok? Or should we do some kind of redirecting?